### PR TITLE
Remove a number of old scrolling widgets:

### PR DESCRIPTION
**Pull request by abarth**
- ScrollableList
 - ScrollableLazyList
 - LazyBlock
 - MaterialList

Clients should use ListView instead.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: examples/flutter_gallery/lib/demo/overscroll_demo.dart [hunk 1]
```diff
@@ -45,8 +45,7 @@ class OverscrollDemoState extends State<OverscrollDemo> {
 
   @override
   Widget build(BuildContext context) {
-    Widget body = new MaterialList( // ignore: DEPRECATED_MEMBER_USE
-      type: MaterialListType.threeLine,
+    Widget body = new Block( // ignore: DEPRECATED_MEMBER_USE
```
Comment (Hixie) [inline examples/flutter_gallery/lib/demo/overscroll_demo.dart]: oh man, you're killing me here. :-P

File: packages/flutter/lib/src/material/list_item.dart [hunk 2]
```diff
@@ -10,6 +10,42 @@ import 'debug.dart';
...
+///  * [MaterialListType]
+///  * [ListItem]
+///  * [kListItemExtent]
+///  * <https://material.google.com/components/lists.html#lists-specs>
+Map<MaterialListType, double> kListItemExtent = const <MaterialListType, double>{
+  MaterialListType.oneLine: 48.0,
+  MaterialListType.oneLineWithAvatar: 56.0,
+  MaterialListType.twoLine: 72.0,
+  MaterialListType.threeLine: 88.0,
+};
```
Comment (Hixie) [inline packages/flutter/lib/src/material/list_item.dart]: These numbers don't really match the actual ground truth in ListItem. I'm not sure it's worth keeping them.

The comment below mentions MaterialList.

**Comment by Hixie** [comment]
All hail the code deletion!

**Comment by Hixie** [comment]
Oh, uh, LGTM also.

File: packages/flutter/lib/src/material/list_item.dart (hunk 2 above)
Comment (abarth) [inline packages/flutter/lib/src/material/list_item.dart]: It's used by two-level list, but in a way that seems quite dodgy.  We can probably remove them.

**Pull request merged**